src/libs/TransactionPreviewUtils.ts

```diff
@@ -0,0 +1,334 @@
+        } else if (isPerDiemRequest(transaction)) {
+            message = translate('common.perDiem');
+        } else if (isScanning) {
+            message = translate('common.receipt');
+        } else if (isBillSplit) {
+            message = translate('iou.split');
+        }
+
+        if (!isCreatedMissing(transaction)) {
+            const created = getFormattedCreated(transaction);
+            const date = DateUtils.formatWithUTCTimeZone(created, DateUtils.doesDateBelongToAPastYear(created) ? CONST.DATE.MONTH_DAY_YEAR_ABBR_FORMAT : CONST.DATE.MONTH_DAY_ABBR_FORMAT);
+            message = `${date} ${CONST.DOT_SEPARATOR} ${message}`;
+        }
+
+        if (isPending(transaction)) {
+            message += ` ${CONST.DOT_SEPARATOR} ${translate('iou.pending')}`;
+        }
+
+        if (hasPendingRTERViolation(violations)) {
+            message += ` ${CONST.DOT_SEPARATOR} ${translate('iou.pendingMatch')}`;
+        }
+
+        if (isMoneyRequestSettled && !iouReport?.isCancelledIOU && !isPartialHold) {
+            message += ` ${CONST.DOT_SEPARATOR} ${getSettledMessage()}`;
+            return message;
+        }
+
+        if (shouldShowRBR && transaction) {
+            return message;
+        }
+
+        if (hasViolationsOfTypeNotice && transaction && !isReportApproved({report: iouReport}) && !isSettled(iouReport?.reportID)) {
+            message += ` ${CONST.DOT_SEPARATOR} ${translate('violations.reviewRequired')}`;
+        } else if (isPaidGroupPolicyExpenseReport(iouReport) && isReportApproved({report: iouReport}) && !isSettled(iouReport?.reportID) && !isPartialHold) {
+            message += ` ${CONST.DOT_SEPARATOR} ${translate('iou.approved')}`;
+        } else if (iouReport?.isCancelledIOU) {
+            message += ` ${CONST.DOT_SEPARATOR} ${translate('iou.canceled')}`;
+        } else if (shouldShowHoldMessage) {
+            message += ` ${CONST.DOT_SEPARATOR} ${translate('violations.hold')}`;
+        }
+        return message;
+    };
+
+    const getDisplayAmountText = (): string => {
+        if (isScanning) {
+            return translate('iou.receiptStatusTitle');
+        }
+
+        if (isFetchingWaypoints && !requestAmount) {
+            return translate('iou.fieldPending');
+        }
+
+        return convertToDisplayString(requestAmount, requestCurrency);
+    };
+
+    const getDisplayDeleteAmountText = (): string => {
+        const iouOriginalMessage: OnyxEntry<OnyxTypes.OriginalMessageIOU> = isMoneyRequestAction(action) ? getOriginalMessage(action) ?? undefined : undefined;
+        return convertToDisplayString(iouOriginalMessage?.amount, iouOriginalMessage?.currency);
+    };
+
+    // const displayAmount = isDeleted ? getDisplayDeleteAmountText() : getDisplayAmountText();
+    return {
+        RBRmessage: getRBRmessage(),
+        displayAmountText: getDisplayAmountText(),
+        displayDeleteAmountText: getDisplayDeleteAmountText(),
+        previewHeaderText: getPreviewHeaderText(),
+    };
```

Comment:
so `createTransactionPreviewText` is a function that returns object witl multiple functions? this is quite confusing.
Looking at it's name I would expect it to return text.